---
title: "R Notebook"
output: html_notebook
---

Here, I analyzed a snRNA-seq dataset of control and 5xFAD mouse brains. 5xFAD is a mouse model of alzheimer's disease.
I performed integration of 2 datasets, because these datasets used different detergents to isolate nuclei, thus creating a batch effect. 
this batch effect must be addressed using an integration method, native to seurat.
i also annotated the cells using the package celldex.

this allowed me to visualize the expression of Ntrk2, my gene of interest, in various cell types of the brain. These include astrocytes, neurons, microglia, and oligodendrocytes.

```{r}
install.packages("tidyverse")
install.packages("Seurat")
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("SingleR")
BiocManager::install("celldex")
BiocManager::install("SeuratData")
remotes::install_github('chris-mcginnis-ucsf/DoubletFinder')

install.packages("devtools")
#for some reason, the celldex package requires this deprecated version of dbplyr
devtools::install_version("dbplyr", version = "2.3.4")

install.packages("pheatmap")
install.packages("plotrix")
install.packages("enrichR")
```


```{r}
library("dplyr")
library("tidyverse")
library("Seurat")
library("ggplot2")
library("beepr")
library("statmod")
library("SingleR")
library("celldex")
library("pheatmap")
source("/Users/xulab/Documents/Rob/RBassFunctions.R")
library("cowplot")
library("ggpubr")
library("plotrix")
library("Hmisc")
library("enrichR")

# install.packages("Rcpp", dependencies = TRUE)
```

```{r}
# fileList <- getFileList("/Users/xulab/Documents/Rob/changran analysis/GSE124952_RAW",
#                     "Sample")
# seu <- createSeuratObjects(fileList, "Samples")
```
# script to perform standard workflow steps to analyze single cell RNA-Seq data

in this dataset, there were 3 separate CSV files that i want to analyze. i am going to load them all into R,
and create a list of seurat objects. all will be analyzed. 


create the seurat object (raw,non-normalized data) 
this function reads CSV files and loads them as seurat objects
```{r}
#create an object, give it a name
#delete genes that aren't expressed in >200 cells
#delete cells that don't have at least 3 genes expressed


readMyCSV <- function(path, sepSemicolon = FALSE) {

if (sepSemicolon == TRUE) {mtxObject <- read.csv(path, sep = ";")}
  else {mtxObject <- read.csv(path)}


#convert the gene names to the row names
rownames(mtxObject) <- mtxObject[[1]]
mtxObject[[1]] <- NULL

#convert to a matrix, then to a sparse matrix, which allows the computer to run faster
mtxObject <- as.matrix(mtxObject)
mtxObject <- as(mtxObject, "sparseMatrix")
# mtxObject

seu <- CreateSeuratObject(counts = mtxObject, project = "project", min.cells = 3, min.features = 200)
return(seu)
}

```




load second seurat object (txt object)
```{r}

readMyTXT <- function(path) {
  mtxObject <- read.delim(path)

  #convert to a matrix, then to a sparse matrix, which allows the computer to run faster
  mtxObject <- as.matrix(mtxObject)
  mtxObject <- as(mtxObject, "sparseMatrix")
  mtxObject

  seu <- CreateSeuratObject(counts = mtxObject, project = "hippo", min.cells = 3, min.features = 200)

  return(seu)
}
```


the first file is hippocampal astrocytes across development
the second is all cell types in hippo
the third is cortical astrocytes
```{r}
seuListRaw <- c(hippo = readMyCSV("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/GSE143758_Admouse_Hippocampus_3TimeCourses_Astrocytes_WT_AD_Aging.csv", sepSemicolon = TRUE),
             large = readMyTXT("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/GSE143758_Admouse_Hippocampus_7m_AllNuclei_UMIcounts.txt"), #i don't care about comparing EZ and NP40 lysis buffer
             ctx = readMyCSV("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/GSE143758_Admouse_Crtx_7-10m_Astrocytes_UMIcounts.csv", sepSemicolon = TRUE))


seuList <- seuListRaw
View(seuList[[2]]@meta.data)
beep(1)

```



```{r}
Idents(seuList[[1]]) <- "genotype"
Idents(seuList[[2]]) <- "genotype"
Idents(seuList[[3]]) <- "genotype"
```






data processing from youtube video:


# 1. QC -------
```{r}

for (i in 1:length(seuList)) {

  seuList[[i]][["percent.mt"]] <- PercentageFeatureSet(seuList[[i]], pattern = "^mt-")


  print(VlnPlot(seuList[[i]], features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3))
  print(FeatureScatter(seuList[[i]], feature1 = "nCount_RNA", feature2 = "nFeature_RNA") +
    geom_smooth(method = 'lm'))

}



```



# 2 and 3. Filtering and normalizing-----------------
take gene expression values of each cell, divide by total gene expression,
multiply by a scaling factor, then log transform. this normalizes expression 
and allows you to compare expression across cells
                          
```{r}
#the default parameters (when you have no parameters specified) are as follows:
#seu <- NormalizeData(seu, normalization.method = "LogNormalize", scale.factor = 10000)
for (i in 1:length(seuList)) {
  seuList[[i]] <- subset(seuList[[i]], subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & 
                          percent.mt < 5)
  seuList[[i]] <- NormalizeData(seuList[[i]])
}


```
                          


# 4. Identify highly variable features --------------



typical method to get highly variable genes
2000 most highly variable genes are saved
```{r}
top10List <- list()
for (i in 1:length(seuList)) {
  seuList[[i]] <- FindVariableFeatures(seuList[[i]], selection.method = "vst", nfeatures = 2000)

  # Identify the 10 most highly variable genes
  # top10 <- head(VariableFeatures(seuList[[i]]), 10)
  top10List <- append(top10List, list(head(VariableFeatures(seuList[[i]]), 10)))

  # plot variable features with and without labels
  plot1 <- VariableFeaturePlot(seuList[[i]])
  # print(LabelPoints(plot = plot1, points = top10, repel = TRUE))
  print(LabelPoints(plot = plot1, points = top10List[[i]], repel = TRUE))
}
```


# 5. Scaling -------------
the data will have unwanted sources of variation. technical noise, such as 
batch effect, differences in cell cycle, etc. 
```{r}
for (i in 1:length(seuList)) {
  all.genes <- rownames(seuList[[i]])
  seuList[[i]] <- ScaleData(seuList[[i]], features = all.genes)
}
```


# 6. Perform Linear dimensionality reduction --------------
```{r}
for (i in 1:length(seuList)) {
  # seuList[[i]] <- RunPCA(seuList[[i]], features = VariableFeatures(object = seuList[[i]]), npcs = 50,ndims.print = 6)
  seuList[[i]] <- RunPCA(seuList[[i]], features = VariableFeatures(object = seuList[[i]]))
  print(VizDimLoadings(seuList[[i]], dims = 1:2, reduction = "pca"))
}
```


# visualize PCA results
```{r}
for (i in 1:length(seuList)) {
  print(seuList[[i]][["pca"]], dims = 1:5, nfeatures = 5)
  DimHeatmap(seuList[[i]], dims = 1:2, cells = 500, balanced = TRUE)
}

# print(seuList[[1]][["pca"]], dims = 1:5, nfeatures = 5)
# DimHeatmap(seuList[[1]], dims = 1, cells = 500, balanced = TRUE)
#   
# print(seuList[[2]][["pca"]], dims = 1:5, nfeatures = 5)
# DimHeatmap(seuList[[2]], dims = 1, cells = 500, balanced = TRUE)
```


# determine dimensionality of the data.
this is also known as a scree plot
```{r}
for (i in 1:length(seuList)) {  
  print(ElbowPlot(seuList[[i]]),ndims = 50,reduction = "pca")
  # print(ElbowPlot(seuList[[i]]))
}
```


# 7. Clustering ------------
```{r}
seuList[[1]] <- FindNeighbors(seuList[[1]], dims = 1:15)
seuList[[2]] <- FindNeighbors(seuList[[2]], dims = 1:15)
seuList[[3]] <- FindNeighbors(seuList[[3]], dims = 1:10)

```


# understanding resolution
0.4 - 1.2 is typically good for single cell datasets with 3k cells. resolution increases for larger datasets
```{r}
# this is the default, it has a lot of clusters
# seu <- FindClusters(seu, resolution = c(0.1,0.3, 0.5, 0.7, 1))
# seuList[[1]] <- FindClusters(seuList[[1]], resolution = c(0.03, 0.05, 0.1, 0.2, 0.3))
# seuList[[2]] <- FindClusters(seuList[[2]], resolution = c(0.03, 0.05, 0.1, 0.2))
# # seuList[[2]] <- FindClusters(seuList[[2]], resolution = c(0.005, 0.008, 0.01))
# # seuList[[2]] <- FindClusters(seuList[[2]], resolution = c(3, 5, 8))
# seuList[[3]] <- FindClusters(seuList[[3]], resolution = c(0.05, 0.1, 0.2, 0.5, 1))

seuList[[1]] <- FindClusters(seuList[[1]], resolution = c(0.1,0.3, 0.5, 0.7, 1))
seuList[[2]] <- FindClusters(seuList[[2]], resolution = c(0.1,0.3, 0.5, 0.7, 1))
seuList[[3]] <- FindClusters(seuList[[3]], resolution = c(0.1,0.3, 0.5, 0.7, 1))


# View(seu@meta.data)

# DimPlot(seuList[[1]], group.by = "RNA_snn_res.0.3", label = TRUE)
# # DimPlot(seuList[[2]], group.by = "RNA_snn_res.0.008", label = TRUE)
# DimPlot(seuList[[2]], group.by = "RNA_snn_res.0.05", label = TRUE)
# DimPlot(seuList[[3]], group.by = "RNA_snn_res.0.5", label = TRUE)

DimPlot(seuList[[1]], group.by = "RNA_snn_res.0.3", label = TRUE)
DimPlot(seuList[[2]], group.by = "RNA_snn_res.0.3", label = TRUE)
DimPlot(seuList[[3]], group.by = "RNA_snn_res.0.3", label = TRUE)
#note that this code makes the identifed clusters the cell identities.
#so does the UMAP code
```


# setting identity of clusters
```{r}
# Idents(seu)
Idents(seuList[[1]]) <- "RNA_snn_res.0.1"
Idents(seuList[[2]]) <- "RNA_snn_res.0.008"
# Idents(seu)
# Idents(object = seu) <- "orig.ident"
```


# non-linear dimensionality reduction --------------
```{r}
# If you haven't installed UMAP, you can do so via reticulate::py_install(packages =
# 'umap-learn')
# this sets the number of groups in your UMAP plot. 
#you want this number to be determined by the variation that your PCA analysis 
#was able to account for
seuList[[1]] <- RunUMAP(seuList[[1]], dims = 1:15)
seuList[[2]] <- RunUMAP(seuList[[2]], dims = 1:15)
seuList[[3]] <- RunUMAP(seuList[[3]], dims = 1:10)
# note that you can set `label = TRUE` or use the LabelClusters function to help label
# individual clusters
DimPlot(seuList[[1]], reduction = "umap")
DimPlot(seuList[[2]], reduction = "umap")
DimPlot(seuList[[3]], reduction = "umap")
#note that this code makes the identifed clusters the cell identities.
#so does the PCA code
```
find all markers of cluster 2
```{r}
marker2List <- list()
for (i in 1:length(seuList)) { 
  markerList <- append(markerList, FindMarkers(seuList[[i]], ident.1 = 2, min.pct = 0.25))
  print(head(markerList[[i]], n = 5))

  # markers <- FindMarkers(seuList[[i]], ident.1 = 2, min.pct = 0.25)
  # print(head(markers, n = 5))
}
```

```{r}

# find all markers distinguishing cluster 5 from clusters 0 and 3
# cluster5.markers <- FindMarkers(seuList[[i]], ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
# head(cluster5.markers, n = 5)

# find markers for every cluster compared to all remaining cells, report only the positive
# ones

seuMarkersList <- list()
for (i in 1:length(seuList)) { 
  seuMarkersList <- append(seuMarkersList, list(FindAllMarkers(seuList[[i]], only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)))
  seuMarkersList[[i]] %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)
}


# seu.markers <- FindAllMarkers(seuList[[1]], only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
# seu.markers %>%
#     group_by(cluster) %>%
#     slice_max(n = 2, order_by = avg_log2FC)

```



visualize how good your clusters are 
```{r}
for (i in 1:length(seuList)) {
seuMarkersList[[i]] %>%
  group_by(cluster) %>%
  top_n(n = 10, wt = avg_log2FC) -> top10List[[i]]
print(DoHeatmap(seuList[[i]], features = top10List[[i]]$gene) + NoLegend())
}

# seu.markers %>%
#   group_by(cluster) %>%
#   top_n(n = 10, wt = avg_log2FC) -> top10
# print(DoHeatmap(seuList[[1]], features = top10$gene) + NoLegend())
# 
# seu.markers2 %>%
#   group_by(cluster) %>%
#   top_n(n = 10, wt = avg_log2FC) -> top10
# print(DoHeatmap(seuList[[2]], features = top10$gene) + NoLegend())
# 
# seu.markers %>%
#   group_by(cluster) %>%
#   top_n(n = 10, wt = avg_log2FC) -> top10
# print(DoHeatmap(seuList[[2]], features = top10$gene) + NoLegend())
```

#check for batch effect ----------------------
```{r}
# seuList[[2]]
p1 <- DimPlot(seuList[[2]], reduction = 'umap', group.by = 'detergent')
print(p1)
```

integrate data
```{r}
splitList <- SplitObject(seuList[[2]], split.by = 'detergent')
for (i in length(splitList)) {
  splitList[[i]] <- NormalizeData(object = splitList[[i]])
  splitList[[i]] <- FindVariableFeatures(object = splitList[[i]])
}

intFeatures <- SelectIntegrationFeatures(object.list = splitList)
anchors <- FindIntegrationAnchors(object.list = splitList)

seuInt <- IntegrateData(anchorset = anchors)
beep(1)
```

scale, run PCA, visualize newly integrated data
```{r}
seuInt <- ScaleData(seuInt)
seuInt <- RunPCA(seuInt)
seuInt <- RunUMAP(seuInt, dims = 1:20)

p2 <- DimPlot(seuInt, reduction = 'umap', group.by = 'detergent')
print(p2)
```

replace dataset
```{r}
seuList[[2]] <- seuInt

# rownames(seuList[[2]])
# rownames(ref)
```

load or save the RDS file
```{r}
# saveRDS(seuList[[1]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ageDataset.rds")
# saveRDS(seuList[[2]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/detergentDataset.rds")
# saveRDS(seuList[[3]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ctxDataset.rds")

# saveRDS(seuList[[1]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ageDatasetw5x.rds")
# saveRDS(seuList[[2]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/detergentDatasetw5x.rds")
# saveRDS(seuList[[3]], file = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ctxDatasetw5x.rds")

#these datasets lack 5xFAD cells!!
# seuList <- list()
# seuList[[1]] <- readRDS("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ageDataset.rds")
# seuList[[2]] <- readRDS("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/detergentDataset.rds")
# seuList[[3]] <- readRDS("/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes/ctxDataset.rds")

# These datasets have 5xFAD and WT cells
seuList <- list()
seuList[[1]] <- readRDS("/Users/xulab/Documents/Rob/GSE143758 Disease associated astrocytes/ageDatasetw5x.rds")
seuList[[2]] <- readRDS("/Users/xulab/Documents/Rob/GSE143758 Disease associated astrocytes/detergentDatasetw5x.rds")
seuList[[3]] <- readRDS("/Users/xulab/Documents/Rob/GSE143758 Disease associated astrocytes/ctxDatasetw5x.rds")
```

cell annotation
```{r}
# GetAssayData(seuList[[2]])
#load reference dataset
ref <- celldex::MouseRNAseqData(ensembl = FALSE, cell.ont = c("all", "nonna", "none"))
# View(as.data.frame(colData(ref)))

#generate cell type labels
countMatrixList <- list()
predLabelsList <- list()
for (i in 1:length(seuList)) {
  # countMatrixList[[i]] <- GetAssayData(seuList[[i]], slot = 'counts', assay = 'RNA')
  # countMatrixList[[i]] <- GetAssayData(seuList[[i]], assay = "RNA")
  countMatrixList[[i]] <- GetAssayData(seuList[[i]])
  
  predLabelsList[[i]] <- SingleR(test = countMatrixList[[i]],
                                ref = ref,
                                labels = ref$label.main)
  # print(predLabelsList[[i]])
}

# View(countMatrixList[[2]])
# View(predLabelsList[[2]])

#test on one seurat object
# seuList2Counts <- GetAssayData(seuList[[2]], slot = 'counts', assay = 'RNA')
# 
# pred <- SingleR(test = seuList2Counts,
#                                 ref = ref,
#                                 labels = ref$label.main)
# pred

```

replace cell type labels with singleR labels
```{r}
for (i in 1:length(seuList)) {
  seuList[[i]]$singleR.labels <- predLabelsList[[i]]$labels[match(rownames(seuList[[i]]@meta.data), rownames(predLabelsList[[i]]))]
  # print(DimPlot(seuList[[i]], reduction = 'umap', group.by = 'singleR.labels'))
}
```

view prediction scores
```{r}
for (i in 1:length(seuList)) { 
  predLabelsList[[i]]
  predLabelsList[[i]]$scores
  plotScoreHeatmap(predLabelsList[[i]])
  print(plotDeltaDistribution(predLabelsList[[i]]))
}
```

compare to unsupervised learning
```{r}
tableList <- list()
for (i in 1:length(seuList)) { 
  tableList[[i]] <- table(Assigned = predLabelsList[[i]]$labels, Clusters = seuList[[i]]$seurat_clusters)
  pheatmap(log10(tableList[[i]]+10), color = colorRampPalette(c("white", "blue"))(10))
}
```

```{r}
for (i in 1:length(seuList)) { 
  View(seuList[[i]]@meta.data)
}
```

compare to your clusters
```{r}
for (i in 1:length(seuList)) {
  Idents(seuList[[i]]) <- "singleR.labels"
  print(DimPlot(seuList[[i]], group.by = "ident"))
}
```

```{r}
hippoUMAP <- DimPlot(seuList[[2]], group.by = "ident")
# ctxUMAP <- DimPlot(seuList[[3]], group.by = "ident")
```


```{r}
# features <- c("Mog", "Aldh1l1", "Cx3cr1", "Rbfox3")
# features <- c("Mog", "Aqp4", "Cx3cr1", "Ncam1")
# features <- c("Mog", "Aldh1l1", "Cx3cr1", "Fas")
# features <- c("Aldh1l1", "Gfap")

features <- "Aldh1l1"

for (i in 1:length(seuList)) {
print(FeaturePlot(object = seuList[[i]],
            features,
            # cols.use = c("grey", "blue"),
            # reduction.use = "umap"
            # ncol = 2 #if there are multiple genes
))
}
# ?FeaturePlot
```

what are marker genes that distinguish WT astrocytes from 5xFAD astrocytes? GO

```{r}
# View(seuList[[2]]@meta.data)
#view WT vs 5x UMAP
# clusters <- DimPlot(seuList[[2]], reduction = "umap", group.by = "singleR.labels")
# condition <- DimPlot(seuList[[2]], reduction = "umap", group.by = "genotype")
# condition|clusters

#create a new column with cell type and condition. make that the cell ident
seuList[[2]]$cellTypeCon <- paste0(seuList[[2]]$singleR.labels, "_", seuList[[2]]$genotype)
View(seuList[[2]]@meta.data)
Idents(seuList[[2]]) <- seuList[[2]]$cellTypeCon

#compare WT astrocytes and 5x astrocytes
abetaResponse <- FindMarkers(seuList[[2]], ident.1 = "Astrocytes_WT", ident.2 = "Astrocytes_5xFAD")

#the enrich database and max genes i have selected here are arbitrary numbers i made up
DEplot <- DEenrichRPlot(seuList[[2]], ident.1 = "Astrocytes_WT", ident.2 = "Astrocytes_5xFAD", balanced = TRUE,
                        assay = "integrated", 
                        enrich.database = "GO_Molecular_Function_2023",
                        # enrich.database = "HomoloGene",
                        max.genes = 1000,
                        num.pathway = 30, return.gene.list = FALSE)
DEplot[[1]]
DEplot[[2]]
# View(seuList[[2]])
```





this will delete the cell types that are irrelevant
```{r}
unique(Idents(seuList[[2]]))

seuList[[2]] <- subset(seuList[[2]], singleR.labels == c("Neurons", "Astrocytes", "Oligodendrocytes", "Microglia"))
# seu <- seuList[[2]]
# Idents(hseu) <- cellTypeFactor
```


```{r}
for (i in 1:length(seuList)) {
  features <- c("Ntrk2")
  print(RidgePlot(seuList[[i]],
          features = features,
          # cols = #colors of groups
          # sort = c("WT 7 week","WT 1 month", "WT 3 month",
          #          "WT 6 month", "WT 8 month","WT 20 month",
          #          "5xFAD 1 month", "5xFAD 3 month", "5xFAD 6 month",
          #          "5xFAD 9 month", "WT 6 month ctx", "5xFAD 6 month ctx"
          #          )
          #change the order depicted
          # group.by = seu@meta.data$group,
          # ncol = 1 #only use if multiple plots are displayed
          ))
}
```


```{r}
features <- c("Ntrk2")
for (i in 1:length(seuList)) {
print(VlnPlot(seuList[[i]],
        features = features,
        # idents = "orig.ident"
        pt.size = 0,
        # adjust = 0.3,
        # slot = seu@meta.data$group
        # cols = c("white", "#53A056") #green
        # sort = c("WT 7 week","WT 1 month", "WT 3 month",
        #            "WT 6 month", "WT 8 month","WT 20 month",
        #            "5xFAD 1 month", "5xFAD 3 month", "5xFAD 6 month",
        #            "5xFAD 9 month", "WT 6 month ctx", "5xFAD 6 month ctx"
        #            )
        # group.by = orig.ident,
        # ncol = 1 #only use if multiple plots are displayed
        ) +
  ggtitle("Ntrk2 expression") +
  xlab(NULL) +
  stat_summary(
    # data = astrocytes,
    # mapping = ,
    fun = mean, fun.min = mean, fun.max = mean,
    geom = "crossbar", linewidth = 0.25,
    width = 0.6, color = "black"
  )
)
}
# Vln
```

```{r}
features <- c("Ntrk2")

hippoNtrk2 <- VlnPlot(seuList[[2]],
        features = features,
        pt.size = 0,
        split.by = "genotype"
        ) +
  ggtitle("Ntrk2 expression") +
  xlab(NULL) +
  stat_summary(
    # data = astrocytes,
    # mapping = ,
    fun = mean, fun.min = mean, fun.max = mean,
    geom = "crossbar", linewidth = 0.25,
    width = 0.6, color = "black"
  ) +
    theme(legend.position = "none")
hippoNtrk2
```

```{r}
View(seuList[[2]]@meta.data)
unique(seuList[[2]]$genotype)
```

```{r}
features <- c("Bdnf")

hippoBdnf <- VlnPlot(seuList[[2]],
        features = features,
        pt.size = 0,
        ) +
  ggtitle("Bdnf expression") +
  xlab(NULL) +
  stat_summary(
    # data = astrocytes,
    # mapping = ,
    fun = mean, fun.min = mean, fun.max = mean,
    geom = "crossbar", linewidth = 0.25,
    width = 0.6, color = "black"
  ) +
  # scale_y_continuous(limits = c(0, 0.25)) +
  theme(legend.position = "none")
hippoBdnf
```


```{r}
features <- c("Bdnf")

ctxBdnf <- VlnPlot(seuList[[3]],
        features = features,
        pt.size = 0,
        ) +
  ggtitle("Bdnf expression") +
  xlab(NULL) +
  stat_summary(
    # data = astrocytes,
    # mapping = ,
    fun = mean, fun.min = mean, fun.max = mean,
    geom = "crossbar", linewidth = 0.25,
    width = 0.6, color = "black"
  ) +
  scale_x_discrete(limits = c("Neurons","Astrocytes", "Epithelial cells")) +
  scale_fill_discrete(breaks = c("Neurons","Astrocytes", "Epithelial cells")) +
  scale_y_continuous(limits = c(0, 0.25)) +
  theme(legend.position = "none")
ctxBdnf
```

```{r}
features <- c("Ntrk2")

ctxNtrk2 <- VlnPlot(seuList[[3]],
        features = features,
        pt.size = 0,
        ) +
  ggtitle("Ntrk2 expression") +
  xlab(NULL) +
  stat_summary(
    # data = astrocytes,
    # mapping = ,
    fun = mean, fun.min = mean, fun.max = mean,
    geom = "crossbar", linewidth = 0.25,
    width = 0.6, color = "black"
  ) +
  scale_x_discrete(limits = c("Neurons","Astrocytes", "Epithelial cells")) +
  scale_fill_discrete(breaks = c("Neurons","Astrocytes", "Epithelial cells")) +
  theme(legend.position = "none")
ctxNtrk2
```



```{r}
# plots <- plot_grid(Ntrk2, Bdnf, labels = c("A", "B"), label_size = 12)

hippoPlots <- ggarrange(hippoBdnf, hippoNtrk2,
                   labels = c("A", "B"),
                   nrow = 2, ncol = 2,
                   heights = c(2,2))
hippoPlots
ggexport(hippoPlots, filename = "Hippocampus Gene Expression.pdf")


ctxPlots <- ggarrange(ctxBdnf, ctxNtrk2,
                   labels = c("A", "B"),
                   nrow = 2, ncol = 2,
                   heights = c(2,2))
ctxPlots
ggexport(ctxPlots, filename = "Cortex Gene Expression.pdf")

UMAPPlots <- ggarrange(hippoUMAP, ctxUMAP,
                   labels = c("A", "B"),
                   nrow = 2, ncol = 1,
                   heights = c(2,2))
UMAPPlots
ggexport(UMAPPlots, filename = "UMAP plots.pdf")

# ggsave("Hippocampus Gene Expression", #filename
#   plot = Ntrk2,
#   # plot =last_plot()#this is an option
#   path = "/Users/xulab/Documents/Rob/changran analysis/GSE143758 Disease associated astrocytes"
# )
```

```{r}
view(seu@meta.data)
unique(seu@meta.data$orig.ident)
```


delete variable list
```{r}
rm(WT)
rm(list = ls())
rm('seu')
# unlist
# new_column_names <- unlist(new_column_names,recursive=F) 
```
